loopd/swapclient_server.go

```diff
@@ -1395,6 +1396,201 @@ func (s *swapClientServer) WithdrawDeposits(ctx context.Context,
 	return &clientrpc.WithdrawDepositsResponse{}, err
 }
 
+// GetStaticAddressSummary returns a summary static address related information.
+// Amongst deposits and withdrawals and their total values it also includes a
+// list of detailed deposit information filtered by their state.
+func (s *swapClientServer) GetStaticAddressSummary(ctx context.Context,
+	req *clientrpc.StaticAddressSummaryRequest) (
+	*clientrpc.StaticAddressSummaryResponse, error) {
+
+	if req.StateFilter != clientrpc.DepositState_UNKNOWN_STATE &&
+		len(req.Outpoints) > 0 {
+
+		return nil, fmt.Errorf("can either filter by state or " +
+			"outpoints")
+	}
+
+	allDeposits, err := s.depositManager.GetAllDeposits()
+	if err != nil {
+		return nil, err
+	}
+
+	return s.depositSummary(
+		ctx, allDeposits, req.StateFilter, req.Outpoints,
+	)
+}
+
+func (s *swapClientServer) depositSummary(ctx context.Context,
+	deposits []*deposit.Deposit, stateFilter clientrpc.DepositState,
+	outpointsFilter []string) (*clientrpc.StaticAddressSummaryResponse,
+	error) {
+
+	var (
+		totalNumDeposits = len(deposits)
+		valueUnconfirmed int64
+		valueDeposited   int64
+		valueExpired     int64
+		valueWithdrawn   int64
+	)
+
+	// Value unconfirmed.
+	utxos, err := s.staticAddressManager.ListUnspent(
+		ctx, 0, deposit.MinConfs-1,
+	)
+	if err != nil {
+		return nil, err
+	}
+	for _, u := range utxos {
+		valueUnconfirmed += int64(u.Value)
+	}
+
+	// Confirmed total values by category.
+	for _, d := range deposits {
+		value := int64(d.Value)
+		switch d.GetState() {
+		case deposit.Deposited:
+			valueDeposited += value
+
+		case deposit.Expired:
+			valueExpired += value
+
+		case deposit.Withdrawn:
+			valueWithdrawn += value
+		}
+	}
+
+	// Deposits filtered by state or outpoints.
+	var clientDeposits []*clientrpc.Deposit
+	if len(outpointsFilter) > 0 {
+		f := func(d *deposit.Deposit) bool {
+			for _, outpoint := range outpointsFilter {
+				if outpoint == d.OutPoint.String() {
+					return true
+				}
+			}
+			return false
+		}
+		clientDeposits = filter(deposits, f)
```

Comment:
I guess we could also filter in the DB but since this is running on the client its not a performance concern. Other thoughts?